pkg/apiserver/filters/auditing.go

```diff
@@ -0,0 +1,42 @@
+package filters
+
+import (
+	"k8s.io/klog"
+	"kubesphere.io/kubesphere/pkg/apiserver/auditing"
+	"kubesphere.io/kubesphere/pkg/apiserver/request"
+	"net/http"
+)
+
+func WithAuditing(handler http.Handler, a auditing.Auditing) http.Handler {
+
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+
+		// When auditing level is LevelNone, request should not be auditing.
+		if !a.Enable() {
+			handler.ServeHTTP(w, req)
+			return
+		}
+
+		e := a.LogRequestObject(req)
+		resp := auditing.NewResponseCapture(w)
+
+		// Create a new goroutine to finish the request, and wait for the response body.
+		// The advantage of using goroutine is that recording the return value of the
+		// request will not affect the processing of the request, even if the auditing fails.
+		go handler.ServeHTTP(resp, req)
```

Comment:
Not true, you still have to wait for the response to return, why not written like following
```suggestion
	 handler.ServeHTTP(resp, req)
	 
	 // function returned, means handler finished processing requests, no need to attach a channel to do the waiting
	 info, ok := request.RequestInfoFrom(req.Context())
	 if !ok {
	    return
	 }
	 go a.LogResponseObject(e, resp, info) // we don't block normal handler processing chain
	 
```